src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnusedCatchParameterShouldBeUnnamedCheck.java

```diff
@@ -0,0 +1,307 @@
+     */
+    private void visitCatchParameter(DetailAST catchAst) {
+        final DetailAST catchParameterAst = catchAst.findFirstToken(TokenTypes.PARAMETER_DEF);
+        final String catchParameterName =
+                catchParameterAst.findFirstToken(TokenTypes.IDENT).getText();
+
+        if (!"_".equals(catchParameterName)) {
+            catchParameter = new CatchParameter(catchParameterName,
+                                    catchAst, catchParameterAst);
+        }
+    }
+
+    /**
+     * Visit ast of type {@link TokenTypes#IDENT}
+     * and check if it is a candidate for a catch parameter usage.
+     *
+     * @param identAST token representing {@link TokenTypes#IDENT}
+     */
+    private void visitIdent(DetailAST identAST) {
+        final boolean isCatchParameter = identAST.getParent().getParent()
+                                                .getType() == TokenTypes.LITERAL_CATCH;
+
+        if (!isCatchParameter
+                && (!TokenUtil.isOfType(identAST.getParent(), UNACCEPTABLE_PARENT_OF_IDENT)
+                        || shouldCheckDotChild(identAST))) {
+            checkIdentifier(identAST);
+        }
+    }
+
+    /**
+     * Check if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     * and should be checked for usage.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     * @return true if the given {@link TokenTypes#IDENT} is a child of a dot operator
+     *     and should be checked for usage.
+     */
+    private static boolean shouldCheckDotChild(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return parent.getType() == TokenTypes.DOT
+                && identAst == parent.getFirstChild();
+    }
+
+    /**
+     * Check the given {@link TokenTypes#IDENT} and
+     * register it as used if it is the catch parameter.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     */
+    private void checkIdentifier(DetailAST identAst) {
+        if (catchParameter != null
+                && !isLeftHandSideValue(identAst)
+                && identAst.getText().equals(catchParameter.getName())) {
+            catchParameter.registerAsUsed();
+        }
+    }
+
+    /**
+     * Check if the given {@link TokenTypes#IDENT} is a left hand side value.
+     *
+     * @param identAst token representing {@link TokenTypes#IDENT}
+     * @return true if the given {@link TokenTypes#IDENT} is a left hand side value.
+     */
+    private static boolean isLeftHandSideValue(DetailAST identAst) {
+        final DetailAST parent = identAst.getParent();
+        return parent.getType() == TokenTypes.ASSIGN
+                && identAst != parent.getLastChild();
+    }
+
+    /**
+     * Log the catch parameter if it is not used after
+     * leaving its scope.
+     *
+     * @param scopeAst ast of the current scope
+     */
+    private void logUnusedCatchParameter(DetailAST scopeAst) {
```

Comment:
This method name doesn't really make sense (we don't always log here), I think we can just add this method body to the leaveToken method.

Reply:
done